Packs/CommonScripts/Scripts/ResolveShortenedURL/ResolveShortenedURL.py

```diff
@@ -1,44 +1,395 @@
+        # or if the resolved URL is the same as the original URL, remove it from `redirect_history`.
+        if (len(redirect_history) >= 2 and redirect_history[-1] == redirect_history[-2]) or \
+                (len(redirect_history) == 1 and redirect_history[0] == original_url):
+            redirect_history.pop()
+
+        return URLUnshorteningData(
+            original_url=original_url,
+            resolved_url=url,
+            service_name=self.service_name,
+            redirect_history=[original_url] + redirect_history,
+            raw_data=raw_data,
+            encountered_error=encountered_error,
+            api_usage=usage_count,
+            api_rate_limit=self.service_rate_limit,
+        )
+
+
+class BuiltInShortener(URLUnshortingService):
+    """A class for unshortening URLs using Python requests."""
+    base_url = ""
+    service_name = "Built-In"
+
+    def resolve_url(self, url: str) -> URLUnshorteningData:
+        encountered_error: bool = False
+        original_url: str = url
+
+        try:
+            response: Response = self._http_request(
+                method="GET",
+                full_url=url,
+                resp_type="response",
+                allow_redirects=False,
+            )
+
+        except Exception:
+            encountered_error = True
+
+        redirect_history: list[str] = []
+
+        while not encountered_error and \
+                (response.is_redirect and (not self.hit_redirect_limit(redirect_history))):
+            url = response.headers["location"]
+            redirect_history.append(url)
+
+            try:
+                response = self._http_request(
+                    method="GET",
+                    full_url=url,
+                    resp_type="response",
+                    allow_redirects=False,
+                )
+
+            except Exception:
+                if not self.hit_redirect_limit(redirect_history):
+                    encountered_error = True
+                break
+
+        return URLUnshorteningData(
+            original_url=original_url,
+            resolved_url=url,
+            service_name=self.service_name,
+            redirect_history=[original_url] + redirect_history,
+            encountered_error=encountered_error,
+        )
+
+
+def unshorten_url(service_name: str, url: str, redirect_limit: int, session_verify: bool = True) -> CommandResults:
+    """
+    Unshorten a shortened URL.
+
+    Args:
+        service_name (str): The service to use for unshortening.
+        url (str): The URL to un-shorten.
+        session_verify (bool): Whether to verify the SSL certificate of the request.
+        redirect_limit (int): A maximum number of recursions to run. Use 0 for unlimited.
+    """
+    error_message: str = "There was an error while attempting to unshorten the final URL in the redirect chain.\n" \
+                         "It is possible that the unshortening process was not fully completed.\n\n"
```

Comment:
```suggestion
    error_message = "There was an error while attempting to unshorten the final URL in the redirect chain.\n" \
                         "It is possible that the unshortening process was not fully completed.\n\n"
```
Why do we need to explicit the type here? It's string and we don't change the type after this